### ISSUE TITLE
Cast of characters (dramatis personae)

### ISSUE THREAD

**Issue by TrevorBurnham** [nyousefi/Fountain#15]
Fountain helpfully provides support for title pages. As explained in the [FAQ](http://fountain.io/faq):

> **Why should Fountain support a title page?**
> One goal of Fountain is that you could write an entire screenplay with it, and easily export and/or print something that a colleague could read and provide feedback.

However, Fountain does not have any syntax for a cast of characters page, a.k.a. _dramatis personae._ While this is not standard for screenplays, it is essential for stage plays. I realize that Fountain is intended primarily for screenwriting, but it seems to me that this one addition would make it a robust tool for stage writing as well.

I propose a syntax like this, which would more generally allow Fountain to generate full pages with two-column tables and a title, which could be useful for call sheets, tech cues, or other informational lists related to a script:
#### Input

```
| Characters
| Claudius | King of Denmark
| Hamlet | Son to the late, and nephew to the present king
```
#### Output

![screen shot 2015-05-31 at 2 22 37 pm](https://cloud.githubusercontent.com/assets/224895/7903122/89e6c4be-07a0-11e5-8a39-b7fa6a0cb446.png)

I'd be happy to work on the implementation and submit a pull request if you think it's a good idea.

(Apologies if this isn't the best forum to request additions to the Fountain spec itself. As far as I could tell, there are no guidelines for feature requests on http://fountain.io.)

**Comment by nyousefi** [comment]
This is a pretty great idea! We're going to talk about how best to do this at our next meeting (I like your suggestion, but we'll see what everyone else thinks).

Thanks!

**Comment by TrevorBurnham** [comment]
Cool! Let me know if there's anything I can do to help.

**Comment by strokirk** [comment]
I'm interested in this as well, since I deal mostly with stage plays, and there are many lists which might need to be generated (dramatis personae, list of scenes, list of songs, etc.). TrevorBurnham's proposed syntax seems good and general purpose - I like it! Although I would write the example as:

```
> Characters <
| Claudius | King of Denmark
| Hamlet | Son to the late, and nephew to the present king
```

The format might even be used to list general column-based information, like this.

```
> Locations <
| **Act 1** | **Act 2**
| Scene 1.1 - A platform before the castle | Scene 2.1 - Polonius' room
```

**Comment by TrevorBurnham** [comment]
@strokirk I considered using the existing `> Centered Text <` syntax, but I think it would be ambiguous here. The syntax should make it clear that the heading is associated with the table, and should therefore be shown at the top of the table's page. (If the table spans multiple pages, then perhaps the heading should be repeated at the top of each of those pages.) To be clear, I'd suggest that a table should have a page break before and after it, but should be able to live anywhere in the script. So let's say you wanted to list the tracks of music that play during the intermission between your first and second acts:

```
KING
Hark! A vagrant!

> End of Act 1 <

| Ode to Joy | Beethoven
| In the Hall of the Mountain King | Edvard Grieg
```

The ambiguity is, would that centered text become the title of the table (and therefore be moved to the top of that page)? Would it depend on whether there's a line of whitespace in between? That's why I think table headings should have a different syntax than other centered text. Preceding the heading with a `|` would clearly convey that the heading is associated with the subsequent table.

**Comment by strokirk** [comment]
@TrevorBurnham True. The only issue I see with starting with a solitary `|` is if you need a table without header (for some reason), and the first row in the table doesn't have a second column. Might be a very rare occurrence. Could perhaps be solved by requiring a tailing `|`, as in `| first, lone column |`.

**Comment by nyousefi** [comment]
This is great, guys. Let me tell you guys where our thinking on this currently sits, with the disclaimer that we're still brainstorming and testing things out. We're really interesting in hearing your thoughts.

We're 100% into this, but we're thinking of this more as lists, rather than tables. Along those lines, we're looking at using "::" as a separator, rather than pipes. So a cast list would look like this:

> Edward :: The Father
> Will :: The Son
> Sandra :: The Mother

etc.

We think it'll look cleaner than pipes, and feels more list-like. In addition, you would be able to add multiple paragraphs on the right hand side by simply omitting the left hand text. That is:

> Edward:: The Father
> :: Edward will be played by the same actor, across multiple time periods.

Would look like this:

![cast-list](https://cloud.githubusercontent.com/assets/159396/8291963/a04892b2-18e1-11e5-8c1e-0469e3e7c725.png)

So far this looks like it'll work well in practice, but let us know what you think.

About headings, the computer programmer in me likes the idea of adding markup for semantics, but I also like the idea of adding as little markup as possible, and to that end I like the idea of just using a Centered element as a heading. I'll bring it up with the guys, though, and see what they think. Feel free to continue making the case for or against. :-)

Nima

**Comment by TrevorBurnham** [comment]
Hmm, so any line containing `::` would be converted into a table row? That feels a little strange to me. All other Fountain syntax features let you look at or near the start of the line to see what kind of feature it is, with the possible exception of transitions (which are generally short).

I suggested `|` because it emulates Markdown's table syntax. Since you want it to look more list-like, did you consider emulating one of Markdown's bullet-point list syntaxes? Maybe not `*`, since that's already used for emphasizing text, but perhaps `+`? And my preference would be to keep `|` as the column delimiter rather than `::`. So, you would have something like this:

```
+ Edward | The Father
+ | Edward will be played by the same actor, across multiple time periods.
+ Will | The Son
```

I can understand not wanting to auto-center single-column table rows, so perhaps it's best to use the centering syntax within the table row syntax to generate a centered heading:

```
+ > Cast of Characters <
+ **Name** | **Played by**
+ Luke Skywalker | Mark Hamill
```

**Comment by strokirk** [comment]
Having worked a little with ReStructuredText lately, I feel that the pipe (`|`) character might still be slightly better as the column delimiter. It's more similar to other markup languages, and the pipe evokes the familiarity of a bordered table.

**Comment by FibreFoX** [comment]
Just being curious, will there be any versions-marker for the file (to inform what features should be expected)? How do your applications behave when having the pipe-char?

**Comment by severdia** [comment]
Is there any resolution on this? I'd love to see a way to add a character list like:

1) Character name
2) Character description
3) Actor name

So I'd like to see this as an option:

```
CAST OF CHARACTERS
Edward: | (Kevin Spacey) The father, Edward, is in his 50s and will be played by the same actor, across multiple time periods.
Will: | (Robert DeNIro) The Son
Sandra: | (Glenn Close) The Mother
```

Or like you have above:

```
CAST OF CHARACTERS
Edward: | (Kevin Spacey) 
| The father, Edward, is in his 50s and will be played by the same actor, across multiple time periods.
Will: | (Robert DeNIro) 
| The Son
Sandra: | (Glenn Close) 
| The Mother
```

The problem with using colons is that standard formatting already uses a colon after character names and that could make things difficult.

**Comment by pierwill** [comment]
Hello! Any news on this issue? If I knew any objective-C I'd give it a show myself 😇

**Comment by severdia** [comment]
I could create a pull request for this if there was any confirmation by admins that it would be accepted.

**Comment by contrapunctus-1** [comment]
I just discovered Fountain and was really excited about trying to use it instead of LaTeX for my theatre scripts, in the hope that it would also be easier for my less-technical fellow artists to read, edit, and write...and then I discovered it doesn't support dramatis personae. 🙁

I could of course make my own compiler with an extension for this, but having it as a part of the standard would be far more preferable!

Could you please let us know where this addition stands?

Thank you for Fountain. ❤️

**Comment by severdia** [comment]
This app supports the syntax described here.

https://apps.apple.com/us/app/scriptigo-pro/id1444743519

**Comment by shaedrich** [comment]
@TrevorBurnham
1. Why not using standard [Markdown tables](https://forum.obsidian.md/t/captions-for-tables-in-markdown/17240/5).
2. If we did that, maybe we could use a [table caption syntax](https://forum.obsidian.md/t/captions-for-tables-in-markdown/17240/5)
  3. Since it is not part of the screenplay, or more precisely the story, I'd suggested also adding a syntax to denote a "credit page" analogous to the title page so that we can separate the appendix from the rest of the screenplay

@nyousefi Since you don't like the idea of using tables and opt for rather using lists, why not utilize [definition lists](https://www.markdownguide.org/extended-syntax#definition-lists). That would also take @TrevorBurnham's concerns about the [Paamayim Nekudotayim](https://en.wikipedia.org/wiki/Scope_resolution_operator#PHP)

@strokirk Instead of centering the text with `><`, we might use [sections](https://fountain.io/syntax/#sections-synopses) since we primarily want to indicate what it is, not how it should be rendered

For those, who are interested in the Scriptigo format: https://scriptigo.com/help/fountain-format/133-improvements-to-the-fountain-format
